src/containers/profile/profile.component.jsx

```diff
@@ -9,19 +11,25 @@ const Profile = (props) => {
   const userFromState = useSelector((state) => state.users.user);
 
   const updateUserInfo = () => {
-    if (!username || !imageAsFile) {
-      alert("Please enter a username and upload an avatar");
-    } else {
-      const userWithUsername = {
-        id: auth.currentUser.uid,
-        username: username,
-        user: userFromState,
-      };
-      addUsernameToStateThunk(userWithUsername);
-      uploadProfileAvatarThunk({ image: imageAsFile, username });
+    usernameCheck(username).then((res) => {
+      if (res) {
+        if (!username || !imageAsFile) {
+          alert("Please enter a username and upload an avatar");
+        } else {
+          const userWithUsername = {
+            id: auth.currentUser.uid,
+            username: username,
+            user: userFromState,
+          };
+          addUsernameToStateThunk(userWithUsername);
+          uploadProfileAvatarThunk({ image: imageAsFile, username });
 
-      //TODO: Need to show user that this is successful or has failed
-    }
+          //TODO: Need to show user that this is successful or has failed
```

Comment:
Is this a nice to have for later?

Reply:
I feel like they get redirected ... thats enough?